code/modules/hydroponics/fermenting_barrel.dm

```diff
@@ -0,0 +1,76 @@
+/obj/structure/fermenting_barrel
+	name = "wooden barrel"
+	desc = "A large wooden barrel. You can ferment fruits and such inside it, or just use it to hold liquid."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "barrel"
+	density = TRUE
+	anchored = FALSE
+	container_type = DRAINABLE | AMOUNT_VISIBLE
+	var/open = FALSE
+	pressure_resistance = 2 * ONE_ATMOSPHERE
+	max_integrity = 300
+	var/speed_multiplier = 1 //How fast it distills. Defaults to 100% (1.0). Lower is better.
+
+/obj/structure/fermenting_barrel/Initialize()
+	create_reagents(300) //Bluespace beakers, but without the portability or efficiency in circuits.
+	. = ..()
+
+/obj/structure/fermenting_barrel/examine(mob/user)
+	. = ..()
+	to_chat(user, "<span class='notice'>It is currently [open?"open, letting you pour liquids in.":"closed, letting you draw liquids from the tap."]</span>")
+
+/obj/structure/fermenting_barrel/proc/makeWine(obj/item/reagent_containers/food/snacks/grown/fruit)
+	if(fruit.reagents)
+		fruit.reagents.trans_to(src, fruit.reagents.total_volume)
+	var/amount = fruit.seed.potency / 4
+	if(fruit.distill_reagent)
+		reagents.add_reagent(fruit.distill_reagent, amount)
+		qdel(fruit)
+		playsound(src, 'sound/effects/bubbles.ogg', 50, TRUE)
+		return
+	else
+		var/data = list()
+		data["names"] = list("[initial(fruit.name)]" = 1)
+		data["color"] = fruit.filling_color
+		data["boozepwr"] = fruit.wine_power
+		if(fruit.wine_flavor)
+			data["tastes"] = list(fruit.wine_flavor = 1)
+		else
+			data["tastes"] = list(fruit.tastes[1] = 1)
+		reagents.add_reagent("fruit_wine", amount, data)
+		qdel(fruit)
+		playsound(src, 'sound/effects/bubbles.ogg', 50, TRUE)
+		return
+
+/obj/structure/fermenting_barrel/attackby(obj/item/I, mob/user, params)
+	var/obj/item/reagent_containers/food/snacks/grown/fruit = I
+	if(istype(fruit))
+		if(!fruit.can_distill)
+			to_chat(user, "<span class='warning'>You can't distill this into anything...</span>")
+			return TRUE
+		else if(!user.transferItemToLoc(I,src))
+			to_chat(user, "<span class='warning'>[I] is stuck to your hand!</span>")
+			return TRUE
+		to_chat(user, "<span class='notice'>You place [I] into [src] to start the fermentation process.</span>")
+		addtimer(CALLBACK(src, .proc/makeWine, fruit), rand(80, 120) * speed_multiplier)
```

Comment:
You probably want a unique timer here

Reply:
What's the difference, if you don't mind me asking?

Reply:
A unique timer where if you create it multiple times it only actually creates 1, useful say when you want an icon to be updated but multiple things could try to update the icon in a single tick and you only want it to run once.

Reply:
If you use a unique addtimer you will only get one makeWine call

Reply:
Yeah

Reply:
ew